### PR TITLE
Cleaning up dead code in System.Net.Mail

### PR DESCRIPTION
**Pull request by Ermiar**
Relative to issue #17905.
Code removed in Common was done after a grep to ensure it is not used anymore. Some resources were kept because they were used by other libraries.
Routing to @karelz as asked.

### CODE REVIEW

**Comment by karelz** [comment]
Thanks @Ermiar! I let @Priya91 @danmosemsft to review the changes.

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop netcoreapp Windows_NT Debug x64 please
@dotnet-bot test Outerloop netcoreapp Ubuntu14.04 Debug x64 please

File: src/System.Net.Mail/src/System/Net/DelegatedStream.cs [hunk 1]
```diff
@@ -14,9 +14,6 @@ internal class DelegatedStream : Stream
         private readonly Stream _stream;
         private readonly NetworkStream _netStream;
 
-        protected DelegatedStream()
```
Comment (danmosemsft) [inline src/System.Net.Mail/src/System/Net/DelegatedStream.cs]: @stephentoub is there any compat impact of removing a protected constructor? eg., user code that calls `base()`?

**Comment by danmosemsft** [comment]
@Priya91 or other owner should signoff

**Comment by stephentoub** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Debug x64 please

File: src/System.Net.Mail/src/System/Net/Mime/EncodedStreamFactory.cs [hunk 2]
```diff
@@ -18,31 +18,31 @@ internal class EncodedStreamFactory
...
-        //get a raw encoder, not for use with header encoding
-        internal IEncodableStream GetEncoder(TransferEncoding encoding, Stream stream)
-        {
-            //raw encoder
-            if (encoding == TransferEncoding.Base64)
-            {
-                return new Base64Stream(stream, new Base64WriteStateInfo());
-            }
+		//get a raw encoder, not for use with header encoding
+		internal IEncodableStream GetEncoder(TransferEncoding encoding, Stream stream)
```
Comment (Priya91) [inline src/System.Net.Mail/src/System/Net/Mime/EncodedStreamFactory.cs]: Undo this file? Seems like it's added tabs instead of spaces.

Comment (Ermiar) [inline src/System.Net.Mail/src/System/Net/Mime/EncodedStreamFactory.cs]: Good catch, I have added a commit to revert back the file.

**Pull request merged**